account_regularization/models/account.py

```diff
@@ -0,0 +1,168 @@
+# -*- encoding: utf-8 -*-
+##############################################################################
+#
+#    OpenERP, Open Source Management Solution
+#    Copyright (c) 2008 Zikzakmedia S.L. (http://zikzakmedia.com)
+#    All Rights Reserved.
+#                       Jordi Esteve <jesteve@zikzakmedia.com>
+#    Copyright (c) 2008 Acysos SL. All Rights Reserved.
+#    $Id$
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU General Public License as published by
+#    the Free Software Foundation, either version 3 of the License, or
+#    (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU General Public License for more details.
+#
+#    You should have received a copy of the GNU General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+from openerp import models, api, fields
+import time
+
+
+class AccountAccount(models.Model):
+    _name = 'account.account'
+    _inherit = 'account.account'
+
+    @api.multi
+    def balance_calculation(self, date=time.strftime('%Y-%m-%d'), periods=[]):
+        acc_set = ",".join(map(str, self.ids))
+        query = self.env['account.move.line']._query_get()
+
+        if not periods:
+            self.env.cr.execute(("SELECT a.id, "
+                                 "COALESCE(SUM(l.debit - l.credit), 0) "
+                                 "FROM account_account a "
+                                 "LEFT JOIN account_move_line l "
+                                 "ON (a.id=l.account_id) "
+                                 "WHERE a.type != 'view' "
+                                 "AND a.id IN (%s) "
+                                 "AND " + query + " "
+                                 "AND a.active "
+                                 "AND l.date <= '%s' "
+                                 "GROUP BY a.id") % (acc_set, date))
+        else:
+            period_ids = ",".join(str(period.id) for period in periods)
+            self.env.cr.execute(("SELECT a.id, "
+                                 "COALESCE(SUM(l.debit - l.credit), 0) "
```

Comment:
Yes I Know, but no time at this moment for this. So I promise to do this in a next PR